crates/bevy_tasks/src/pollable_task.rs

```diff
@@ -0,0 +1,31 @@
+use crate::Task;
+use async_channel::{Receiver, TryRecvError};
+
+/// A pollable task whose result readiness can be checked in system functions
+/// on every frame update without blocking on a future
+#[derive(Debug)]
+pub struct PollableTask<T> {
+    receiver: Receiver<T>,
+    // this is to keep the task alive
+    _task: Task<()>,
+}
+
+impl<T> PollableTask<T> {
+    pub(crate) fn new(receiver: Receiver<T>, task: Task<()>) -> Self {
+        Self {
+            receiver,
+            _task: task,
+        }
+    }
+
+    /// poll to see whether the task finished
+    pub fn poll(&self) -> Option<T> {
+        match self.receiver.try_recv() {
+            Ok(value) => Some(value),
+            Err(try_error) => match try_error {
+                TryRecvError::Empty => None,
+                TryRecvError::Closed => panic!("todo"),
```

Comment:
- [ ] I'm not sure what to do here. Any suggestions? Panic? Return a Result instead of an Option?

Reply:
I feel pretty strongly that this should return a Result instead.

Reply:
Or a custom enum `PollResult<T> { Finished(T), Pending, Error }`?
Actually pending is not really an error case, so it would feel a bit weird to allow `pollable.poll()?` there. `Result<Option<T>>` feels too verbose.

Reply:
Async functions panic when polling after they have finished.

Reply:
Sounds good. I'm convinced by `bjorn3`'s argument: we should try to follow convention here.